src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -203,7 +204,12 @@ public static IEnumerable<DType> GetPrimitiveTypes()
         /// the expected type exactly, however, so it is necessary to set this value for a single aggregate DType
         /// and not for the individual field types within.
         /// </summary>
-        public bool AreFieldsOptional { get; set; } = false;
+        public bool AreFieldsOptional { get; set; } = false;        
+        
+        /// <summary>
+        /// Check if DType Union/Coerce should be executed under PowerFxV1 compat rules.
+        /// </summary>
+        public Features Features { get; set; }
```

Comment:
DType shouldn't have to know about features; you can pass the 'usePowerFxV1CompatibilityRules' flag to the functions that need to know about this